NEWS.md

```diff
@@ -78,6 +78,8 @@ Standard library changes
 ------------------------
 
 * `gcdx(0, 0)` now returns `(0, 0, 0)` instead of `(0, 1, 0)` ([#40989]).
+* New `hascodepoint(c::AbstractChar)` function returns
+  whether `codepoint(c)` will succeed ([#54393]).
```

Comment:
This description seems a bit misleading since `Base.codepoint` is not implemented for `AbstractChar`, but `Base.hascodepoint` is

Reply:
If it's not implemented for any concrete `AbstractChar` type, then that type will be broken.

For example, `UInt32(c)` is [defined in terms of `codepoint`](https://github.com/JuliaLang/julia/blob/5f7bfc0a8374a902b723870262ffe7ddfbc44e77/base/char.jl#L50), as is `Char(c)`.   If `Char(c)` fails, then `isless` will fail because that [works via `Char(c)`](https://github.com/JuliaLang/julia/blob/5f7bfc0a8374a902b723870262ffe7ddfbc44e77/base/char.jl#L228).  And if `isless` fails, then lots of other methods that do `<= 'x'` will fail, such as [show](https://github.com/JuliaLang/julia/blob/5f7bfc0a8374a902b723870262ffe7ddfbc44e77/base/char.jl#L291).